frontend/js/tests/user/stats/UserReports.test.tsx

```diff
@@ -1,173 +1,144 @@
+    const handlers = apiRoutes.map((route) => {
+      return http.get(route, async (path) => {
+        return HttpResponse.json({});
       });
     });
+    server = setupServer(...handlers);
+    server.listen();
+  });
+  afterEach(() => {
+    queryClient.cancelQueries();
+    queryClient.clear();
+  });
 
-    describe("syncStateWithUrl", () => {
-      it("calls getURLParams once", () => {
-        const wrapper = shallow<UserReports>(<UserReports {...props} />);
-        const instance = wrapper.instance();
-
-        instance.getURLParams = jest.fn();
-        instance.syncStateWithURL();
-
-        expect(instance.getURLParams).toHaveBeenCalledTimes(1);
-      });
-
-      it("sets state correcty", () => {
-        const wrapper = shallow<UserReports>(<UserReports {...props} />);
-        const instance = wrapper.instance();
-
-        instance.getURLParams = jest.fn().mockImplementationOnce(() => "month");
-        instance.syncStateWithURL();
+  afterAll(() => {
+    server.close();
+  });
 
-        expect(wrapper.state("range")).toEqual("month");
-      });
+  describe("UserReports", () => {
+    it("Check if component is being rendered", () => {
+      renderWithProviders(
+        <RouterProvider router={router} />,
+        {},
+        {
+          wrapper: reactQueryWrapper,
+        },
+        false
+      );
+
+      // Check if the component is rendered correctly
+      expect(screen.getByTestId("User Reports")).toBeInTheDocument();
+
+      // The week pill should be selected by default
+      const weekPill = screen.getByTestId("range-week");
+      expect(weekPill).toHaveClass("active");
     });
 
-    describe("getURLParams", () => {
-      it("gets default parameters if none are provided in the URL", () => {
-        const wrapper = shallow<UserReports>(<UserReports {...props} />);
-        const instance = wrapper.instance();
-
-        window.location = {
-          href: "https://foobar.org",
-        } as Window["location"];
-        const range = instance.getURLParams();
+    it("should navigate on range change", async () => {
+      renderWithProviders(
+        <RouterProvider router={router} />,
+        {},
+        {
+          wrapper: reactQueryWrapper,
+        },
+        false
+      );
 
-        expect(range).toEqual("week");
-      });
+      const rangePill = screen.getByTestId("range-month");
+      expect(rangePill).toBeInTheDocument();
 
-      it("gets parameters if provided in the URL", () => {
-        const wrapper = shallow<UserReports>(<UserReports {...props} />);
-        const instance = wrapper.instance();
+      // Click on the month pill
+      await user.click(rangePill);
 
-        window.location = {
-          href: "https://foobar.org?range=year",
-        } as Window["location"];
-        const range = instance.getURLParams();
-
-        expect(range).toEqual("year");
-      });
+      // Check if the URL is updated
+      expect(mockSearchParam).toEqual({ range: "month" });
```

Comment:
Here I wonder if you could also test  that the query client has received a query.
For example using [queryClient.getQueryState](https://tanstack.com/query/latest/docs/reference/QueryClient/#queryclientgetquerystate) with the right queryKey, which has a `dataUpdatedAt` or `dataUpdateCount` properties that could be used to ensure there is something reacting to the pill being clicked

Reply:
Fixed